### PR TITLE
Add nk_***_get_scroll and nk_***_set_scroll for groups, windows, and popups

### PR DESCRIPTION
**Pull request by keharriso**
Split #783 into two. This half of the changes adds `nk_***_get_scroll` and `nk_***_set_scroll` for groups, windows, and popups. It also bumps the version to 4.01.0

### CODE REVIEW

**Comment by dumblob** [comment]
Thanks @keharriso, it looks good to me. It's clear, readable and simple. I didn't test it though.
@vurtun feel free to give it a short test and I think we can merge it.

**Comment by irtusb** [comment]
what if I only want vertical scroll offset? nk_window_get_scroll() should accept NULL to ignore one or both args

**Comment by keharriso** [comment]
> what if I only want vertical scroll offset? nk_window_get_scroll() should accept NULL to ignore one or both args. Same remark for the other two getters.

Done, thanks for the suggestion.

**Comment by borodust** [comment]
Any chance this would be merged anytime soon?

**Comment by dumblob** [comment]
Thanks for the reminder. Please rebase this pull on top of the current version and I'll merge it.

**Comment by dumblob** [comment]
Thanks for rebasing - there are though some conflicts (even though GitHub doesn't show any conflicts for some reason) - see e.g. the content of https://github.com/vurtun/nuklear/blob/master/demo/x11_rawfb/Makefile and the diff in this pull.

Could you please check the pull for such cases? Thank you for understanding.

**Comment by keharriso** [comment]
I'm afraid I don't see the difference.
https://github.com/vurtun/nuklear/blob/master/demo/x11_rawfb/Makefile
and
https://github.com/keharriso/nuklear/blob/get-set-scroll/demo/x11_rawfb/Makefile
look exactly the same to me.

**Comment by borodust** [comment]
I wonder if upstream master was force pushed or @keharriso fork got corrupted in some way. I've rebased @keharriso's branch onto vurtun/master and after a bit of skipping commits i've got nice history back, or so it seems: https://github.com/vurtun/nuklear/compare/master...borodust:get-set-scroll

File: doc/nuklear.html [hunk 1]
```diff
@@ -2258,6 +2307,9 @@
    - [x]: Major version with API and library breaking changes
    - [yy]: Minor version with non-breaking API and library changes
    - [zz]: Bug fix version with no direct changes to API
+- 2019/06/23 (4.01.0) - Added nk_***_get_scroll and nk_***_set_scroll for groups, windows, and popups
+- 2019/06/12 (4.00.3) - Fix panel background drawing bug
```
Comment (borodust) [inline doc/nuklear.html]: I also got these lines after rebasing. Are those correct?

**Comment by dumblob** [comment]
@keharriso Travis has updated their distro and we didn't update the script so far (any pulls welcome - my bet is, that the ppa is not any more required as everything is in the default repositories).

>I'm afraid I don't see the difference.

There is no difference, but this shouldn't happen and I didn't check all the rest of the pull because if such thing happens, it must be cleared up first.

Actually I've checked the branch `get-set-scroll` from @borodust and can confirm, that this one is totally clear and is exactly what I would expect in the diffs. @keharriso any suggestions what would be your preferred way to proceed?

**Comment by keharriso** [comment]
I'm pretty sure I just did the rebase wrong. I don't have much experience with that.

**Comment by keharriso** [comment]
I just force pushed the branch from @borodust

File: doc/nuklear.html (hunk 1 above)
Comment (keharriso) [reply doc/nuklear.html]: I believe they reflect the changes in CHANGELOG

Comment (borodust) [reply doc/nuklear.html]: Ah, that's a generated file, my bad.

File: nuklear.h [hunk 2]
```diff
@@ -2609,6 +2643,8 @@ NK_API struct nk_rect nk_layout_space_rect_to_local(struct nk_context*, struct n
 /// nk_group_scrolled_offset_begin  | Start a new group with manual separated handling of scrollbar x- and y-offset
 /// nk_group_scrolled_begin         | Start a new group with manual scrollbar handling
 /// nk_group_scrolled_end           | Ends a group with manual scrollbar handling. Should only be called if nk_group_begin returned non-zero
+// nk_group_get_scroll              | Gets the scroll offset for the given group
```
[the remark this answers is not in the document]
Comment (keharriso) [reply nuklear.h]: Alright, hopefully that commit fixes the slashes.

**Comment by borodust** [comment]
Thanks!

**Pull request merged**